.github/workflows/lint.yml

```diff
@@ -11,19 +11,17 @@ jobs:
   golangci:
     name: golangci-lint
     runs-on: ubuntu-latest
-    timeout-minutes: 6
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2.1.4
+        with:
+          go-version: 1.16
       - uses: technote-space/get-diff-action@v5
+        id: git_diff
         with:
           PATTERNS: |
             **/**.go
             go.mod
             go.sum
-      - uses: golangci/golangci-lint-action@master
-        with:
-          # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
-          version: v1.39
-          args: --timeout 10m
-          github-token: ${{ secrets.github_token }}
+      - name: run go linters
+        run: go run github.com/golangci/golangci-lint/cmd/golangci-lint run --out-format=tab ${{ env.GIT_DIFF }}
```

Comment:
I'm linting only changed files (NOT all files). I think this speedups the flow.

Reply:
Why not just run the make target?

Reply:
this will be nice if another pr causes issues for a PR that doesnt touch it. That being said i could see linting getting messy if prs are merged without fully checking things

Reply:
> Why not just run the make target?

Because `make` job lints all files. Linting only changed files speeds up the process.  I'm fixing "all" issues in other PR.

> That being said i could see linting getting messy if prs are merged without fully checking things

This requires a social consensus and an admin power - in that case we should have a task to fix that later.

Reply:
Moving forward - I'm OK to change it to always lint everything. My motivation was that an author should be responsible for "his" changes in a PR, and not being blocked by someone else changes which get in using super powers.
Also speeding up the CI is a nice thing.

Reply:
What in this command ensures only changed files are linted? I don't see it.

Reply:
I like in PRs this running for changed files and on master it runs for everything. That way we could see if other things are failing

Reply:
@alexanderbez it is the `${{ env.GIT_DIFF }}` part

Reply:
@marbar3778 We have only actions for PRs. I'm not sure if we can make an action for a commit to master.... This sounds more like a status. But that sounds like a new feature: _master dashoboard_

Reply:
@alexanderbez  : how about breaking down the `make lint` and add `make lint-go` and use `$GIT_DIFF` there?